File: recursive/fib.hs
-- ghc -O -O2 -optc-O3 fib.hs
fib :: Int -> Int
fib 0 = 1
fib 1 = 1
fib n = fib(n-1) + fib(n-2)

main = do
    print (fib 40)

File: recursive/Fib.java
class Fib {
	static int fib(int n) {
		if (n < 2)
			return 1;
		return fib(n-2) + fib(n-1);
	}

	public static void main(String[] args) {
		System.out.println(fib(40));
	}
}
